axum/src/extract/path/de.rs

```diff
@@ -823,4 +834,38 @@ mod tests {
             ErrorKind::UnsupportedType { name: "(u32, u32)" }
         );
     }
+
+    #[test]
+    fn test_untagged_enum() {
+        #[derive(serde::Serialize, serde::Deserialize, PartialEq, Eq, Debug)]
+        #[serde(untagged)]
+        enum Main {
+            A(A),
+            B(B),
+        }
+
+        #[derive(serde::Serialize, serde::Deserialize, PartialEq, Eq, Debug)]
+        enum A {
+            A1,
+            A2,
+        }
+
+        #[derive(serde::Serialize, serde::Deserialize, PartialEq, Eq, Debug)]
+        enum B {
+            B1,
+            B2,
+        }
+
+        let url_params = create_url_params(vec![("a", "A2")]);
+        assert_eq!(
+            Main::deserialize(PathDeserializer::new(&url_params)).unwrap(),
+            Main::A(A::A2)
+        );
+
+        let url_params = create_url_params(vec![("a", "B2")]);
+        assert_eq!(
+            Main::deserialize(PathDeserializer::new(&url_params)).unwrap(),
+            Main::B(B::B2)
+        );
```

Comment:
Thanks for adding a test! Would you wanna add an integration test as well [here](https://github.com/tokio-rs/axum/blob/main/axum/src/extract/path/mod.rs#L412)? Then we'll know that it actually works end-to-end.

Reply:
Done!